.NET/Microsoft.Recognizers.Text.DateTime/Utilities/MergedParserUtil.cs

```diff
@@ -397,6 +398,16 @@ public static string DetermineSourceEntityType(string sourceType, string newType
             var resolutionPast = GenerateResolution(type, pastResolutionStr, mod);
             var resolutionFuture = GenerateResolution(type, futureResolutionStr, mod);
 
+            if (type.Equals(Constants.SYS_DATETIME_SET, StringComparison.Ordinal) && (config.Options & DateTimeOptions.TasksMode) != 0)
```

Comment:
It's better to check for the mode flag first, as much as possible. This makes the code easier to read and potentially saves a few cycles in the default mode skipping unnecessary checks there.

Reply:
code modified as per suggestion.